gdnative-core/src/geom/basis.rs

```diff
@@ -19,8 +25,89 @@ impl Basis {
         unsafe { std::mem::transmute::<sys::godot_basis, Self>(c) }
     }
 
-    pub fn identity() -> Basis {
+    /// The basis that will flip something along the **X Axis** when used in a transformation.
+    pub const fn flip_x() -> Basis {
+        Basis::from_diagonal(Vector3::new(-1.0, 1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the **Y axis** when used in a transformation.
+    pub const fn flip_y() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, -1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the **Z axis** when used in a transformation.
+    pub const fn flip_z() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, 1.0, -1.0))
+    }
+
+    /// Creates a Basis from the given [`Vector3`](./type.Vector3.html)
+    pub const fn from_diagonal(p_diag: Vector3) -> Basis {
         Basis {
+            elements: [
+                Vector3::new(p_diag.x, 0.0, 0.0),
+                Vector3::new(0.0, p_diag.y, 0.0),
+                Vector3::new(0.0, 0.0, p_diag.z),
+            ],
+        }
+    }
+
+    /// Creates a Basis from 3 [`Vectors3`](./type.Vector3.html)s
+    pub fn from_elements(elements: [Vector3; 3]) -> Self {
+        Self { elements }
+    }
+
+    /// Creates a rotation matrix.
+    pub fn from_euler(euler: Vector3) -> Self {
+        let mut b = Basis::default();
+        b.set_euler_yxz(&euler);
+        b
+    }
+
+    /// Rotation matrix from axis and angle.
+    ///
+    /// See [https://en.wikipedia.org/wiki/Rotation_matrix#Rotation_matrix_from_axis_angle]()
+    pub fn from_axis_angle(&axis: &Vector3, phi: f32) -> Self {
+        assert!(
+            axis.length().approx_eq(&1.0),
+            "The axis Vector3 must be normalized."
+        );
+
+        let mut basis = Basis::default();
+        let [x, y, z] = &mut basis.elements;
+
+        let axis_sq = Vector3::new(axis.x * axis.x, axis.y * axis.y, axis.z * axis.z);
+        let cosine = phi.cos();
+
+        x.x = axis_sq.x + cosine * (1.0 - axis_sq.x);
+        y.y = axis_sq.y + cosine * (1.0 - axis_sq.y);
+        z.z = axis_sq.z + cosine * (1.0 - axis_sq.z);
+
+        let sine = phi.sin();
+        let t = 1.0 - cosine;
+
+        let mut xyzt = axis.x * axis.y * t;
+        let mut zyxs = axis.z * sine;
+        x.y = xyzt - zyxs;
+        y.x = xyzt + zyxs;
+
+        xyzt = axis.x * axis.z * t;
+        zyxs = axis.y * sine;
+        x.z = xyzt + zyxs;
+        z.x = xyzt - zyxs;
+
+        xyzt = axis.y * axis.z * t;
+        zyxs = axis.x * sine;
+        y.z = xyzt - zyxs;
+        z.y = xyzt + zyxs;
+
+        basis
+    }
+
+    /// The identity basis.
+    ///
+    /// Identical to calling [`Basis::default()`](struct.Basis.html#method.default). This constant can be used to make your code clearer.
+    pub const fn identity() -> Basis {
```

Comment:
Maybe this should be moved to the top, being the "simplest" constructor?

Also, "This constant can be used to make your code clearer." is not needed in the doc-comments. This is not a constant, and we're not teaching people how to code. We expect our users to know what makes their code clearer.